Formula/f/flexiblas.rb

```diff
@@ -0,0 +1,110 @@
+class Flexiblas < Formula
+  desc "BLAS and LAPACK wrapper library with runtime exchangable backends"
+  homepage "https://www.mpi-magdeburg.mpg.de/projects/flexiblas"
+  url "https://csc.mpi-magdeburg.mpg.de/mpcsc/software/flexiblas/flexiblas-3.4.4.tar.xz"
+  sha256 "f3b4db7175f00434b1ad1464c0fd004f9b9ddf4ef8d78de5a75382a1f73a75dd"
+  license all_of: [
+    "LGPL-3.0-or-later",
+    "LGPL-2.1-or-later", # libcscutils/
+    "BSD-3-Clause-Open-MPI", # contributed/
+  ]
+  head "https://gitlab.mpi-magdeburg.mpg.de/software/flexiblas-release.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "gcc" # for gfortran
+  depends_on "openblas"
+
+  on_macos do
+    depends_on "libomp" => :build
+  end
+
+  fails_with :gcc do
+    version "11"
+    cause "Need to build with same GCC as GFortran for LTO"
+  end
+
+  def blas_backends
+    backends = %w[OPENBLASOPENMP NETLIB]
+    on_sonoma :or_newer do
+      backends.unshift("APPLE")
+    end
+    on_ventura :or_older do
+      backends << "APPLE"
+    end
+    backends
+  end
+
+  def install
+    # Remove -flat_namespace usage
+    inreplace "src/fallback_blas/CMakeLists.txt", " -flat_namespace\"", '"'
+
+    # Add HOMEBREW_PREFIX path to default searchpath to allow detecting separate formulae
+    inreplace "CMakeLists.txt",
+              %r{(FLEXIBLAS_DEFAULT_LIB_PATH "\${CMAKE_INSTALL_FULL_LIBDIR}/\${flexiblasname}/)"},
+              "\\1:#{HOMEBREW_PREFIX}/lib/${flexiblasname}\""
+
+    args = %W[
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+      -DFLEXIBLAS_DEFAULT=#{blas_backends.first}
+      -DSYSCONFDIR=#{etc}
+      -DEXAMPLES=OFF
+      -DTESTS=OFF
+    ]
+
+    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    # Let brew manage linking/deleting config files that are intended to be read-only
+    (prefix/"etc").install etc/"flexiblasrc.d"
```

Comment:
I will need to adjust this for source-builds to other formulae to use directory. Maybe something like following:
```suggestion
    (prefix/"etc/flexiblasrc.d").install ((etc/"flexiblasrc.d").children - etc_before)
```